internal/actions/sync_branch.go

```diff
@@ -435,29 +403,29 @@ func syncBranchContinue(
 		msgRebaseResult(rebase)
 	}
 
-	// Finish setting the new trunk for the branch
-	if opts.Continuation.NewTrunk != "" {
-		var err error
-		branch, err = syncBranchUpdateNewTrunk(tx, branch, opts.Continuation.NewTrunk)
-		if err != nil {
-			return nil, err
-		}
+	// Finish setting the parent for the branch
+	branch, err = syncBranchUpdateParent(tx, branch, opts.Continuation)
+	if err != nil {
+		return nil, err
 	}
-
 	return &SyncBranchResult{*rebase, nil, branch}, nil
 }
 
-func syncBranchUpdateNewTrunk(
+func syncBranchUpdateParent(
 	tx meta.WriteTx,
 	branch meta.Branch,
-	newTrunk string,
+	continuation *SyncBranchContinuation,
 ) (meta.Branch, error) {
```

Comment:
Actually, this has a bug. I'll fix that. We should addremove the child/parents properly.

Reply:
Fixed. Now this adds a child when the parent is changed and the parent is not trunk.